### PR TITLE
Uas 32+

### PR DESCRIPTION
**Pull request by pm-dattaprasad-mundada**

### CODE REVIEW

File: src_new/controllers/UAS/executionQueue.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+var UTIL = require("../../util.js");
+
+exports.EexecutionQueue = function(){
+	var Phoenix = window.Phoenix;
```
Comment (PubMatic-OpenWrap) [inline src_new/controllers/UAS/executionQueue.js]: can we take Phoenix object in argument than accessing global namespace ?

File: src_new/controllers/UAS/executionQueue.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	this.internalQ = [];
+
+	if( ! UTIL.isUndefined(Phoenix) && ! UTIL.isUndefined(Phoenix.EQ) ){
+		this.internalQ = Phoenix.EQ;
+	}
+
+	this.push = function( theFunction ){
+		var Phoenix = window.Phoenix;
+		if(typeof theFunction === "function"){
```
Comment (PubMatic-OpenWrap) [inline src_new/controllers/UAS/executionQueue.js]: use UTIL.isFunction

File: src_new/controllers/UAS/slotObject.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+	return this.getKeysFromSlotPairs(this.slotExtraParameters, key);
+};
+
+// return array of all common targeting keys
+SlotObject.prototype.getExtraPatameterKeys = function(){
+	return this.getKeysFromSlotPairs(this.slotExtraParameters);
+};
+
+// common function - set key value at slots level
+SlotObject.prototype.setSlotPairs = function(slotVar, key, value){
```
Comment (PubMatic-OpenWrap) [inline src_new/controllers/UAS/slotObject.js]: This function need not be exposed

File: src_new/controllers/UAS/slotObject.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+	var i, len;
+
+	// check type of value , always maintain an array of values against a key
+	if( ! slotVar.hasOwnProperty(key) ){
+		slotVar[key] = [];
+	}
+
+	if(UTIL.isArray(value)){
```
Comment (PubMatic-OpenWrap) [inline src_new/controllers/UAS/slotObject.js]: use Util function to loop

File: src_new/controllers/UAS/slotObject.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+		returnValue = slotVar[key];
+	}
+
+	return returnValue;
+};
+
+SlotObject.prototype.getKeysFromSlotPairs = function(slotVar){
+	var returnArray = [];
+
+	for(var key in slotVar){
```
Comment (PubMatic-OpenWrap) [inline src_new/controllers/UAS/slotObject.js]: Use Util functions

File: src_new/controllers/UAS/uas.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+		initPhoenix(win);
+	}, 500);
+}
+/* start-test-block */
+exports.createPubMaticNamespace = createPubMaticNamespace;
+/* end-test-block */
+
+//todo: change variable names
+function generateBCUID(win){
+	var c = win.document.createElement("script"),
```
Comment (PubMatic-OpenWrap) [inline src_new/controllers/UAS/uas.js]: use Util function

File: webpack.config.js [hunk 7]
```diff
@@ -13,6 +18,24 @@ module.exports = {
...
+        loaders: [
+            {
+                test: /owt.js$/,
+                include: /(src_new)/,
+                loader: StringReplacePlugin.replace({
+                  replacements: [
+                    {
+                      pattern: /%%PATH_TO_CONTROLLER%%/g,
+                      replacement: function (match, p1, offset, string) {
+                        return controllerPaths[conf.pwt.adserver];
```
Comment (PubMatic-OpenWrap) [inline webpack.config.js]: Default controller should be DFP

File: src_new/controllers/UAS/phoenix.js [hunk 8]
```diff
@@ -0,0 +1,529 @@
...
+
+    //todo: what to do if adunit is targeted at all sizes ?
+    //		empty array should be passed []
+    //		but then what should be passed ?
+    case 'adSizes':
+      if( dataAttributeFunctionMap.hasOwnProperty( dataAttribute ) ){
+        // temp will contain an array of sizes
+        temp = theSlot[ dataAttributeFunctionMap[ dataAttribute ] ]();
+        len = temp.length;
+        for(i=0; i<len; i++){
```
Comment (PubMatic-OpenWrap) [inline src_new/controllers/UAS/phoenix.js]: use util function to loop

File: src_new/controllers/UAS/phoenix.js [hunk 9]
```diff
@@ -0,0 +1,529 @@
...
+var EQ = require("./executionQueue.js");
+var SO = require("./slotObject.js");
+
+var serevr_path = "ae.pubmatic.com/ad?";
+
+function combineGlobalTargetings(commonTargetings) {
+  var targetinKey,
+    output = '';
+
+  for(targetingKey in commonTargetings){
```
Comment (PubMatic-OpenWrap) [inline src_new/controllers/UAS/phoenix.js]: use Util function to loop

File: src_new/controllers/UAS/phoenix.js [hunk 10]
```diff
@@ -0,0 +1,529 @@
...
+
+  queryString.push(
+    'req_type=' + req_type,
+    'sec=' + window.PWT.secure,
+    'res_format=2',
+    //Callback function
+    'cback=window.parent.Phoenix.callback', // todo: window.parent ??  we may need to rethink
+    //Random Number
+    'rndn='			+ Math.random(),
+    'purl='			+ encodeURIComponent(window.PWT.pageURL),
```
Comment (PubMatic-OpenWrap) [inline src_new/controllers/UAS/phoenix.js]: use Util function

File: src_new/controllers/UAS/phoenix.js [hunk 11]
```diff
@@ -0,0 +1,529 @@
...
+    if(queryParams.hasOwnProperty(key) && testParams.hasOwnProperty(key)){
+      queryString.push(key + '=' + queryParams[key]);
+    }
+  }
+
+  return window.PWT.protocol + serevr_path + queryString.join('&');
+}
+
+function createFriendlyIframeAndTriggerAdServerCall( addIframeToElementID, adServerRequestCall ){
+  var addIframeToElement,
```
Comment (PubMatic-OpenWrap) [inline src_new/controllers/UAS/phoenix.js]: is there a Util function to simplify this ?

File: src_new/controllers/UAS/phoenix.js [hunk 12]
```diff
@@ -0,0 +1,529 @@
...
+
+  	// sets common targeting applicable to all slots
+  	this.setCommonTargeting = function(key, value){
+  		var i, len;
+  		// check type of value , always maintain an array of values against a key
+  		if( ! this.commonTargetings.hasOwnProperty(key) ){
+  			this.commonTargetings[key] = [];
+  		}
+
+  		if(UTIL.isArray(value)){
```
Comment (PubMatic-OpenWrap) [inline src_new/controllers/UAS/phoenix.js]: use Util functions

File: src_new/controllers/UAS/phoenix.js (hunk 10 above)
Comment (pm-dattaprasad-mundada) [reply src_new/controllers/UAS/phoenix.js]: We use util.getMetaInfo(win) once in owt.js which calculates & returns big object & we assign its output to window.PWT.XX . So I think current implementation is right

**Pull request closed without merging**